lib/src/deploy.rs

```diff
@@ -340,7 +340,7 @@ pub(crate) async fn prepare_for_pull(
     imgref: &ImageReference,
     target_imgref: Option<&OstreeImageReference>,
 ) -> Result<PreparedPullResult> {
-    let ostree_imgref = &OstreeImageReference::from(imgref.clone());
+    let ostree_imgref = &OstreeImageReference::from(imgref.canonicalize());
```

Comment:
It *might* be worth adding a `tracing::debug!` or so in the places where we canonicalize just so we could track when this happens.

Or perhaps we should start logging more to the systemd journal actually.

Reply:
I've added a trace.  
As to what is logged to the systemd journal, I think this needs some decisions to take.  I don't believe the `canonicalize()` function would be worth journaling, since you're probably already logging the actual imgref during the switch.